File: java/pattern/src/facade/RegisterFacade.java
package facade;

public class RegisterFacade {
    public void register() {
        Step1 s1 = new Step1();
        s1.register();
        Step2 s2 = new Step2();
        s2.register();
        Step3 s3 = new Step3();
        s3.register();
    }
}

File: php/facade/RegisterFacade.php
<?php

class RegisterFacade
{
    public function register()
    {
        $s1 = new Step1();
        $s1->register();

        $s2 = new Step2();
        $s2->register();

        $s3 = new Step3();
        $s3->register();
    }
}